src/nft/NftMetadataProvider/index.tsx

```diff
@@ -77,14 +77,19 @@ export const metadataCallBatcher = (() => {
   };
 })();
 
-export function useNftMetadata(contract: string, tokenId: string): NFTResource {
+// DEPRECATED, use useNftResource
+export function useNftMetadata(
+  contract: string | undefined,
+  tokenId: string | undefined
+): NFTResource {
   const { cache, loadNFTMetadata } = useContext(NftMetadataContext);
 
-  const key = getNftKey(contract, tokenId);
+  const key = contract && tokenId ? getNftKey(contract, tokenId) : "";
```

Comment:
this will fix metadata to return error when a falsy contract/tokenid was given (can happen sometimes apparently)